src/jit/codegenarm64.cpp

```diff
@@ -736,31 +730,25 @@ void CodeGen::genRestoreCalleeSavedRegistersHelp(regMaskTP regsToRestoreMask, in
     static_assert_no_msg(REGSIZE_BYTES == FPSAVE_REGSIZE_BYTES);
     int spOffset = lowestCalleeSavedOffset + regsToRestoreCount * REGSIZE_BYTES;
 
-    if (genSaveFpLrWithAllCalleeSavedRegisters)
-    {
-        // TODO: always save int regs higher than float, to be consistent?
-        regMaskTP maskRestoreRegsFloat = regsToRestoreMask & RBM_ALLFLOAT;
-        regMaskTP maskRestoreRegsInt   = regsToRestoreMask & ~maskRestoreRegsFloat;
+    // Save integer registers at higher addresses than floating-point registers.
 
-        // Restore in the opposite order of saving.
+    regMaskTP maskRestoreRegsFloat = regsToRestoreMask & RBM_ALLFLOAT;
+    regMaskTP maskRestoreRegsInt   = regsToRestoreMask & ~maskRestoreRegsFloat;
 
-        if (maskRestoreRegsInt != RBM_NONE)
-        {
-            int spIntDelta = (maskRestoreRegsFloat != RBM_NONE) ? 0 : spDelta; // should we delay the SP adjustment?
-            genRestoreCalleeSavedRegisterGroup(maskRestoreRegsInt, spIntDelta, spOffset);
-            spOffset -= genCountBits(maskRestoreRegsInt) * REGSIZE_BYTES;
-        }
+    // Restore in the opposite order of saving.
 
-        if (maskRestoreRegsFloat != RBM_NONE)
-        {
-            // If there is any spDelta, it must be used here.
-            genRestoreCalleeSavedRegisterGroup(maskRestoreRegsFloat, spDelta, spOffset);
-            // No need to update spOffset since it's not used after this.
-        }
+    if (maskRestoreRegsInt != RBM_NONE)
+    {
+        int spIntDelta = (maskRestoreRegsFloat != RBM_NONE) ? 0 : spDelta; // should we delay the SP adjustment?
```

Comment:
I do not understand that comment. Do you mean to do sp adjustment here like:
```
if (maskRestoreRegsInt != RBM_NONE)
{
  genRestoreCalleeSavedRegisterGroup(maskRestoreRegsInt, spDelta, spOffset);
  spDelta = 0;
}
if (maskRestoreRegsFloat != RBM_NONE)
{
  genRestoreCalleeSavedRegisterGroup(maskRestoreRegsFloat, spDelta, spOffset);
}

```

Is it for correctness or for readability?

Reply:
Note that I haven't changed this logic; I've just picked one path and un-indented it.

In this case, we need to do the opposite of the "save" case. In the "save" case, we make the spDelta adjustment with the first thing saved -- either the float registers, if any, or the int registers if no float regs. When restoring, we make the spDelta adjustment *last* -- so we need to see if there are any float regs that we'll restore later, and defer the spDelta adjustment if there are.

Reply:
Got it, I misunderstood the comment.